### ISSUE TITLE
Figure linewidth not documented and not visible when saved

### ISSUE THREAD

**Issue by tdpetrou** [matplotlib/matplotlib#9370]
### Bug report

**Bug summary**

When creating a figure it is possible to use `linewidth` along with `edgecolor` to create a border around the figure. The linewidth parameter is not shown [in the documentation](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure). There is no corresponding `set_linewidth` figure method.

Also, when saving to a file, this border is missing.

I am using Jupyter notebook.

**Code for reproduction**
```python
fig = plt.figure(linewidth=20, edgecolor='r')
plt.plot([5,6])
plt.savefig('/Users/Ted/Desktop/figlw.png')
```

**Actual outcome (in notebook)**

<img width="394" alt="screen shot 2017-10-11 at 11 32 37 am" src="https://user-images.githubusercontent.com/7226751/31450707-1b68a7a6-ae78-11e7-83cb-e0d57d28abb2.png">

**Expected outcome**

**Matplotlib version**

  * Operating system: OSX
  * Matplotlib version: 2.1
  * Matplotlib backend (`module://ipykernel.pylab.backend_inline`):
  * Python version: 3.6
  * Jupyter version (if applicable): 5.0
  * Other libraries: 

Updated through conda

**Comment by WeatherGod** [comment]
Uhmmm. I have no clue how that is working at all for you. It isn't
documented because I don't think any of the devs ever conceived that
feature...

**Comment by jklymak** [comment]
I can confirm this as the behaviour.  `plt.show()` has the nice red box, printing does not.  

It is documented, at least on master in the Figure class:

```python

def __init__(self,
                 figsize=None,  # defaults to rc figure.figsize
                 dpi=None,  # defaults to rc figure.dpi
                 facecolor=None,  # defaults to rc figure.facecolor
                 edgecolor=None,  # defaults to rc figure.edgecolor
                 linewidth=0.0,  # the default linewidth of the frame
                 frameon=None,  # whether or not to draw the figure frame
                 subplotpars=None,  # default to rc
                 tight_layout=None,  # default to rc figure.autolayout
                 ):
        """
        *figsize*
            w,h tuple in inches

        *dpi*
            Dots per inch

        *facecolor*
            The figure patch facecolor; defaults to rc ``figure.facecolor``

        *edgecolor*
            The figure patch edge color; defaults to rc ``figure.edgecolor``

        *linewidth*
            The figure patch edge linewidth; the default linewidth of the frame

```

no idea why its not rendering when a png is saved...

**Comment by tdpetrou** [comment]
Ok,now I'm confused. I see it [documented here at matplotlib.figure.Figure](https://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html)

**Comment by anntzer** [comment]
fwiw, only occurs in notebook inline backend, not notebook backend

**Comment by WeatherGod** [comment]
Huh... I stand corrected. I just never figured that was a thing. I learn
something new all the time.

If this is happening only in the inline backend, then this is a bug on
ipython/jupyter's part, not ours. They maintain that backend.

**Comment by anntzer** [comment]
Well.  It's still an issue for us because using e.g. qt5agg you get the same discrepancy.

This is because the figure edgecolor is overridden by the `savefig.edgecolor` rcParam at save time.  Setting `rcParams["savefig.edgecolor"] = "r"` fixes the issue for qt5agg (I can see why this rcParam (or rather its companion, rcParams["savefig.facecolor"]) used to exist back when the UI interface defaulted to a gray background but you'd want to save figures with a white background, but as usual these days it seems to be more a pain point than anything and I'd suggest either deprecating them or **at least** switching the default from "w" to something that says "don't touch it").

But even setting savefig.edgecolor doesn't fix it for jupyter inline mode.

**Comment by jklymak** [comment]
I reproduced in `MacOSX` backend.

**Comment by QuLogic** [comment]
I believe this is fixed by #15111.
